src/params/params.py

```diff
@@ -32,20 +32,22 @@
 # logger config
 logging.basicConfig(
     level=LOG_LEVEL,
-    format='%(asctime)s[%(levelname)s] %(message)s',
-    datefmt='[%Y-%m-%d %H:%M:%S]',
-    handlers=[
-        logging.StreamHandler(sys.stdout)
-    ]
+    format="%(asctime)s[%(levelname)s] %(message)s",
+    datefmt="[%Y-%m-%d %H:%M:%S]",
+    handlers=[logging.StreamHandler(sys.stdout)],
 )
 
 # Device
-DEVICE = torch.device("cuda" if torch.cuda.is_available() else "cpu")
-logging.log(logging.INFO, f'Using device: {DEVICE}')
-
-
-
-
-
-
-
+if torch.cuda.is_available():
+    DEVICE = torch.device("cuda")
+    print("Using CUDA")
+elif torch.backends.mps.is_available():
```

Comment:
I added support for `mps`. It's interesting that the `TAN_03` fails for me:
```
Traceback (most recent call last):
  File "/Users/wojciechachtelik/Documents/studia/doktorat/Adaptive_PINN_1D/main.py", line 19, in <module>
    train_PINN(
  File "/Users/wojciechachtelik/Documents/studia/doktorat/Adaptive_PINN_1D/src/runners/adaptive_PINN.py", line 58, in train_PINN
    stage_convergence_data = train_model(nn_approximator=pinn, loss_fn=loss_fn, device=params.DEVICE, learning_rate=params.LEARNING_RATE,
  File "/Users/wojciechachtelik/Documents/studia/doktorat/Adaptive_PINN_1D/src/base/pinn_core.py", line 90, in train_model
    loss = loss_fn(pinn=nn_approximator)
  File "/Users/wojciechachtelik/Documents/studia/doktorat/Adaptive_PINN_1D/src/problems/tan_03.py", line 28, in compute_loss
    assert (boundary_right == self.range[1]).all().item(), f"Last point not on a boundary: {boundary_right}"
AssertionError: Last point not on a boundary: tensor([[1.5708]], device='mps:0', grad_fn=<ViewBackward0>)
```
Probably the reason is the fact that `mps` doesn't support `torch.float64`.

Reply:
I haven't experienced the issue 
We can either debug that together or I can assist you with all needed computation with my setup and GPU